Add night mode

keerifox opened the pull request:
![2021-02-15_21-53-02](https://user-images.githubusercontent.com/29272208/108093518-b5e37780-7075-11eb-8711-2b3c295b3c1f.gif)


Night mode is the new default option (even if the OS prefers light mode), however the user can toggle it in the sidebar at any time. The setting is automatically stored in local storage just like any other setting. The light theme is considered "legacy" and I really don't want to have it as default, but if someone really really wants to use it, they can.

Fixes #304 (single account wallets showing an icon to hide the account)

Also fixes several UI/UX issues, see attached commits for details

keerifox committed:
dark mode (work in progress)

keerifox committed:
night mode toggle and fixes

keerifox committed:
qr code icon as svg, fix icon colors on light theme

keerifox committed:
dark mode fixes and address book page improvements

keerifox committed:
apply dark/light class to document body, support modals

keerifox committed:
fix incorrect rep list placeholder in offline mode

keerifox committed:
fix qr code for receive being generated late, and never in offline mode

keerifox committed:
hide rep and balance sections of account details in offline mode

keerifox committed:
list recommended reps above local avoid, add label for avoid reps

keerifox committed:
fix 'Add New Contact' caption when editing existing address book entry

keerifox committed:
dark mode support for alerts and notifs, style fixes

keerifox committed:
fix header and sortable colors

keerifox committed:
fix 'Add Representative' header when editing known rep

keerifox committed:
prevent editing ids of existing address book and rep entries

make sure no other known rep is using the same name when saving changes
allow "saving" address book entries even if the name was left unchanged
add dark mode styles for input placeholders and disabled inputs

keerifox committed:
fix lint

keerifox committed:
fix style of wallet backup alert, fix alert icon shrinking

keerifox committed:
truncate page headings

keerifox committed:
fix password input style being overridden

keerifox committed:
add some space at the bottom of main content column

keerifox committed:
keep wallet setup option background transparent

keerifox committed:
improved modals for resetting wallet and local storage

keerifox committed:
fix sidebar nav color affected by main section style

keerifox committed:
style fixes for new wallet page

keerifox committed:
toggle light/dark logo on welcome page based on style

keerifox committed:
hide manage wallet nav link when no wallet is configured

keerifox committed:
fix text color of select input with invalid option selected

keerifox committed:
wallet setup page style improvements

BitDesert commented on src/app/app.component.html:
```diff
@@ -226,21 +226,49 @@
+          <div class="uk-width-1-3 uk-text-center">
+            <a class="support-nault" routerLink="/send" [queryParams]="{ to: donationAccount }" routerLinkActive="active">
+              <div class="button">
+                <a class="footer-link donation-icon"><span uk-icon="icon: heart; ratio: 1.2;"></span></a>
```
Don't know if a solid white heart icon looks better, but unfortunately UIkit does not have a solid white one.

keerifox replied:
it does, but only on hover. added that

BitDesert commented on src/app/app.component.html:
```diff
@@ -226,21 +226,49 @@
+            <div class="icons-row">
+              <a href="https://github.com/Nault/Nault" uk-icon="icon: github-alt; ratio: 1.2;" uk-tooltip title="View our GitHub" class="footer-link" target="_blank" rel="noopener noreferrer"></a>
+              <a href="https://discord.nanocenter.org" uk-tooltip title="Join our community on Discord" class="footer-link discord-icon" target="_blank" rel="noopener noreferrer">
+                <svg width="28" height="28" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 245 240"><path class="st0" d="M104.4 103.9c-5.7 0-10.2 5-10.2 11.1s4.6 11.1 10.2 11.1c5.7 0 10.2-5 10.2-11.1.1-6.1-4.5-11.1-10.2-11.1zM140.9 103.9c-5.7 0-10.2 5-10.2 11.1s4.6 11.1 10.2 11.1c5.7 0 10.2-5 10.2-11.1s-4.5-11.1-10.2-11.1z"/><path class="st0" d="M189.5 20h-134C44.2 20 35 29.2 35 40.6v135.2c0 11. ...
```
Can we move the inline SVGs to individual files at `src/assets/img/` ? Better for browser caching.

keerifox replied:
easier to change colors of inline svgs, would want to avoid using `-webkit-mask` unless absolutely neccessary

could revisit outside the pr

BitDesert commented on src/app/app.component.html:
```diff
@@ -226,21 +226,49 @@
+            <div [class]="{'dark-mode-click-area': true, 'enabled': !settings.settings.lightModeEnabled}" (click)="toggleLightMode()">
+              <div class="dark-mode-area">
+                <div class="dark-mode-switch">
+                  <svg width="16" height="16" version="1.1" viewBox="0 0 160 160" xmlns="http://www.w3.org/2000/svg">
```
Another SVG here (see above)

BitDesert commented on src/less/components/dark-mode.less:
```diff
@@ -0,0 +1,458 @@
+@dark-mode-background-1: #19191B;
+@dark-mode-background-2: #1B1B20;
+@dark-mode-background-3: #1D1D22;
+@dark-mode-background-4: #22222A;
```
I've seen this color at some places, it got a blue/purple tint to it, is this intended?
Tried out `#1F1F24` from the sidebar too but I don't know what I like better.

keerifox replied:
yeah it's intentional, i might try tweaking the sidebar colors more in the future but looks good enough already

BitDesert commented on src/less/nault-theme.less:
```diff
@@ -25,29 +25,84 @@ h1, h2, h3, h4, h5, .uk-text-lead, .uk-button, .uk-alert, .uk-description-list d
 	}
 }
 
 .uk-description-list dt {
-    font-weight: 700;
-    color: @global-color;
+	font-weight: 700;
+	color: @global-color;
```
Did the spacing increase? Is this a local or repo setting?

keerifox replied:
changed it to tabs to be consistent with my older .less files

BitDesert commented on src/app/app.component.html:
```diff
@@ -226,21 +226,49 @@
+            </div>
+            <div class="section-caption">Community</div>
+          </div>
+          <div class="uk-width-1-3 uk-text-center dark-mode-column">
```
Just wondered if we could add a sun icon when you choose light mode 😄 
Something like https://codepen.io/jsndks/pen/qEXzOQ

keerifox replied:
thought about it but would prefer to keep the moon at all times, esp. with the setting named night mode

keerifox committed:
fill the heart icon on hover

keerifox committed:
update sidebar style to better match new design

keerifox committed:
increase click area of the support button

keerifox committed:
custom scrollbar for the main section

keerifox committed:
hide sidebar under 940px width rather than under 640px

fix color of horizontal separator on account details page in dark mode
fix weird scrollbars appearing on document body
fix main section scrollbar going deeper than the window height on small viewport

keerifox committed:
new wallet style in the sidebar

keerifox committed:
fix color balance of previous commit
